src/ApparentHorizons/YlmSpherepack.cpp

```diff
@@ -527,37 +529,54 @@ YlmSpherepack::first_and_second_derivative(
 template <typename T>
 YlmSpherepack::InterpolationInfo<T>::InterpolationInfo(
     const size_t m_max, const std::vector<double>& pmm,
-    const std::array<T, 2>& target)
-    : cos_theta(cos(target[0])),
+    const std::array<T, 2>& target_points)
+    : cos_theta(cos(target_points[0])),
       cos_m_phi(m_max + 1),
       sin_m_phi(m_max + 1),
       pbar_factor(m_max + 1),
-      num_points(get_size(target[0])) {
-  // do not use auto here, the blaze expression templates will create a dangling
-  // reference that is not caught
-  const T sin_theta = sin(target[0]);
-  const T& phi = target[1];
+      num_points(get_size(target_points[0])) {
+
+  const auto& theta = target_points[0];
+  const auto& phi = target_points[1];
+
+  ASSERT(get_size(theta) == get_size(phi),
+         "Size mismatch for thetas and phis: " << get_size(theta) << " and "
+                                               << get_size(phi));
+  if (m_max < 2) {
+    ERROR("Must use m_max>=2, not m_max=" << m_max);
+  }
+
+  // `DataVectors` for working. `pbar_factor` is guaranteed to be at least size
+  // 3 as demanded by the `YlmSpherePack` constructor
+  T sinmphi{};
+  T cosmphi{};
+  auto& alpha = pbar_factor.at(0);
+  auto& beta = pbar_factor.at(1);
+  auto& deltasinmphi = pbar_factor.at(2);
 
   // Evaluate cos(m*phi) and sin(m*phi) by numerical recipes eq. 5.5.6
   {
-    cos_m_phi[0] = make_with_value<T>(phi, 1.);
-    sin_m_phi[0] = make_with_value<T>(phi, 0.);
-    const T beta = sin(phi);
-    const T alpha = 2.0 * square(sin(0.5 * phi));
-    T sinmphi = make_with_value<T>(phi, 0.);
-    T cosmphi = make_with_value<T>(phi, 1.);
+    alpha = 2.0 * square(sin(0.5 * phi));
+    beta = sin(phi);
+    sinmphi = make_with_value<T>(num_points, 0.);
+    cosmphi = make_with_value<T>(num_points, 1.);
+    cos_m_phi[0] = make_with_value<T>(num_points, 1.);
+    sin_m_phi[0] = make_with_value<T>(num_points, 0.);
     for (size_t m = 1; m < m_max + 1; ++m) {
-      const T deltacosmphi = alpha * cosmphi + beta * sinmphi;
-      const T deltasinmphi = alpha * sinmphi - beta * cosmphi;
-      cosmphi -= deltacosmphi;
+      deltasinmphi = alpha * sinmphi - beta * cosmphi;
+      cosmphi -= alpha * cosmphi + beta * sinmphi;
       sinmphi -= deltasinmphi;
       cos_m_phi[m] = cosmphi;
       sin_m_phi[m] = sinmphi;
     }
   }
 
+  T& sin_theta = sinmphi;
+  T& sinmtheta = cosmphi;
+  sin_theta = sin(theta);
+  sinmtheta = make_with_value<T>(num_points, 1.);
```

Comment:
do you need to do the extra allocation here? Can't you just do `sinmtheta = 1` and reuse the allocation from before?

Reply:
I did not consider that that would allocate and did not realize the assignment operator was overloaded in this way. Thanks, I fixed it